server/api.js

```diff
@@ -84,4 +85,88 @@ router.get("/quizes",(req, res)=>{
 	});
 });
 
+router.get("/quizes/search",(req, res)=>{
+	const client = getClient();
+	client.connect(()=>{
+		const db = client.db("quiz");
+		const collection = db.collection("quizes");
+		const quiz = { quiz_name:req.query.quiz_name };
+		collection.findOne(quiz,(err, questions)=>{
+			if(err){
+				res.status(500).send(err);
+			}else{
+				res.status(200).send(questions);
+			}
+		});
+	});
+});
+
+router.get("/results/search",(req, res)=>{
+	const client = getClient();
+	client.connect(()=>{
+		const db = client.db("quiz");
+		const collection = db.collection("results");
+		const quiz = { name:req.query.name };
+		collection.find(quiz).toArray((err, results)=>{
+			if(err){
+				res.status(500).send(err);
+			}else{
+				res.status(200).send(results);
+			}
+		});
+	});
+});
+router.get("/results",(req, res)=>{
+	const client = getClient();
+	client.connect(()=>{
+		const db = client.db("quiz");
+		const collection = db.collection("results");
+		collection.find().toArray((err, results)=>{
+			if(err){
+				res.status(500).send(err);
+			}else{
+				res.status(200).send(results);
+			}
+		});
+	});
+});
+
+router.post("/results",(req, res)=>{
+	const client = getClient();
+	client.connect(()=>{
+		const db = client.db("quiz");
+		const collection = db.collection("results");
+		const question = req.body;
+
+		collection.insertOne(question,(err, result)=>{
+			if(err){
+				res.status(500).send(err);
+			}else{
+				res.status(200).send(result.ops[0]);
+			}
+		});
+	});
+});
+
+router.get("/question/:id",(req, res)=>{
+	const client = getClient();
+	client.connect(()=>{
+		const db = client.db("quiz");
+		const collection = db.collection("questions");
+
+		const questionId = req.params.id;
+		if (!mongodb.ObjectID.isValid(questionId)) {
+			return res.status(400).json("the ID is not valid");
+		}
+		const id = mongodb.ObjectID(questionId);
+		collection.findOne(id,(err, questions)=>{
```

Comment:
Nitpick: `questions` should be `question`, given we're doing a `findOne`

Reply:
absolutely correct!